specification/cognitiveservices/data-plane/ComputerVision/preview/v3.1-preview.2/Ocr.json

```diff
@@ -0,0 +1,500 @@
+            "StorageException",
+            "InvalidPageRange"
+          ],
+          "x-ms-enum": {
+            "name": "ComputerVisionErrorCodes",
+            "modelAsString": true
+          }
+        },
+        "message": {
+          "description": "A message explaining the error reported by the service.",
+          "type": "string"
+        },
+        "requestId": {
+          "description": "A unique request identifier.",
+          "type": "string"
+        }
+      }
+    },
+    "ImageUrl": {
+      "type": "object",
+      "required": [
+        "url"
+      ],
+      "properties": {
+        "url": {
+          "description": "Publicly reachable URL of an image.",
+          "type": "string"
+        }
+      }
+    }
+  },
+  "parameters": {
+    "OcrDetectionLanguage": {
+      "name": "language",
+      "in": "query",
+      "description": "The BCP-47 language code of the text in the document. Currently, only English ('en'), Dutch ('nl'), French ('fr'), German ('de'), Italian ('it'), Portuguese ('pt'), and Spanish ('es') are supported. Read supports auto language identification and multi-language documents, so only provide a language code if you would like to force the documented to be processed as that specific language.",
+      "required": false,
+      "default": "en",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "x-ms-enum": {
+        "name": "OcrDetectionLanguage",
+        "modelAsString": true
+      },
+      "enum": [
+        "en",
+        "es",
+        "fr",
+        "de",
+        "it",
+        "nl",
+        "pt"
+      ]
+    },
+    "ImageUrl": {
+      "name": "ImageUrl",
+      "in": "body",
+      "required": true,
+      "x-ms-parameter-location": "method",
+      "x-ms-client-flatten": true,
+      "description": "A JSON document with a URL pointing to the image that is to be analyzed.",
+      "schema": {
+        "$ref": "#/definitions/ImageUrl"
+      }
+    },
+    "ImageStream": {
+      "name": "Image",
+      "in": "body",
+      "required": true,
+      "x-ms-parameter-location": "method",
+      "description": "An image stream.",
+      "schema": {
+        "type": "object",
+        "format": "file"
+      }
+    },
+    "Endpoint": {
+      "name": "Endpoint",
+      "description": "Supported Cognitive Services endpoints.",
+      "x-ms-parameter-location": "client",
+      "required": true,
+      "type": "string",
+      "in": "path",
+      "x-ms-skip-url-encoding": true
+    },
+    "Pages": {
+      "name": "Pages",
+      "in": "query",
+      "description": "Custom page numbers for multi-page documents(PDF/TIFF), input the number of the pages you want to get OCR result. For a range of pages, use a hyphen. Separate each page or range with a comma or space.",
+      "required": false,
+      "x-ms-parameter-location": "method",
+      "type": "array",
+      "items": {
+        "type": "string",
+        "pattern": "(^[0-9]+-[0-9]+$)|(^[0-9]+$)"
```

Comment:
This would not allow you to specify open ended ranges (e.g. `8-` to get all pages starting with page 8)

Reply:
Actually I think we're agreed with that we don't have open ended ranges in Pages in the meeting. We only have two formats: one is "2,3,4", another is "2-4".

Reply:
If that is what the service supports, then this is (a) correct representation. If the service does support open ranges, then we'd have to update the pattern.